pkg/cmd/kitimport/util.go

```diff
@@ -163,20 +163,35 @@ func getEditorName() (string, error) {
 }
 
 // extractRepoFromURL attempts to normalize a string or URL into a repository name as is used on GitHub and Huggingface.
-// Returns an error we cannot automatically handle the input URL/string.
-//   - https://example.com/segment1/segment2 --> segment1/segment2
-//   - 'organization/repository'             --> 'organization/repository'
-func extractRepoFromURL(rawUrl string) (string, error) {
+// Returns the normalized repository path (org/repo), the kind ("model" or "dataset"), and an error if parsing fails.
+//   - https://huggingface.co/org/repo --> (org/repo, "model", nil)
+//   - https://huggingface.co/datasets/org/repo --> (org/repo, "dataset", nil)
+//   - 'organization/repository' --> (organization/repository, "model", nil)
+//   - 'datasets/organization/repository' --> (organization/repository, "dataset", nil)
+func extractRepoFromURL(rawUrl string) (string, string, error) {
```

Comment:
These changes change the purpose of this function, making it specific for handling huggingface repositories.

```go
// extractRepoFromURL attempts to normalize a string or URL into a repository name as is used on GitHub and Huggingface.
```
vs
```
Returns the normalized repository path (org/repo), the kind ("model" or "dataset"),
```
-- the 'kind' here only makes sense in a huggingface context, not GitHub (or any Git repository, for that matter).